src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -677,6 +728,19 @@ internal static MethodInfo GetClientMethod(Type t, string serverDefinedName, Typ
             return methodInfo;
         }
 
+
+        private static ODataTypeInfo GetODataTypeInfo(Type type)

```

Comment:
Get...() should return "null" if the type is not in the dict?

Reply:
Get is intended to add to dict if a type is not present and then return it

Reply:
Just use GetOrAdd on the concurrent dictionary.